docs/configuration.md

```diff
@@ -428,6 +428,37 @@ insert newlines.
 > You might be interested in the [Vertical Multiline](#vertical-multiline)
 > section.
 
+### `newlines.source`
+
+This parameter controls the general approach to line breaks, and whether to take into
+account existing newlines in the source. The default value (if the parameter is not
+specified) is the _classic_, original way. Below are the alternatives.
+
+> These alternatives are EXPERIMENTAL and might change in the future without
+> regard to any `edition` settings, until fully released (and this message deleted).
+
+#### `newlines.source=keep`
```

Comment:
Ideally, we should add code examples, but it can be done later

Reply:
i considered it... since it covers way too many cases, a bit hard. suggestions?

Reply:
I would pick at least a few illustrative examples from unit tests

For example:

```
val nested =
  promiseIntercept(
    new Actor {
      def receive = {
        case _ =>
      }
    }
  )(result)

class Sample {
  def mkFunc(): String => Boolean = {
    if(flag) {
      v => true
    } else {
      v => false
    }
  }
}
```